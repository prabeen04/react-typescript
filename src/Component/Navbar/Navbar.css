.transparent-bg{
    background-color: transparent !important;
}

.logo{
    color: red;
    font-family: Lobster;
    font-weight: 600;
    font-size: 30px !important;
}
.logo:hover{
    color: red;
}

.nav-menu{
    color: red;
    font-family: Lobster;
    font-size: 20px !important;
}
.nav-menu:hover{
    color: teal;
}

.active-menu{
    color: teal;
}
  /* On screens that are 992px or less, set the background color to blue */
@media screen and (max-width: 992px) {
    .my-nav{
        box-shadow: 0px 1px 6px -4px  rgb(245, 50, 1);
    }
}


/* On screens that are 600px or less, set the background color to olive */
@media screen and (max-width: 600px) {
    .my-nav{
        box-shadow: 0px 1px 6px -4px  rgb(245, 50, 1);
    }
    .navbar-toggler-icon{
        background: red !important;
        color: red !important;
    }
}