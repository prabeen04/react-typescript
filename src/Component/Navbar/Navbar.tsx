import * as React from 'react'

export default function Navbar() {
  return (
    <>
      
    </>
  )
}
