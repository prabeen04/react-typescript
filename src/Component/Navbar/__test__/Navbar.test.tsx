import * as React from 'react';
import { shallow } from "enzyme";
import Navbar from '../Navbar';

describe('<Navbar />', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<Navbar/>)
        expect(wrapper).toMatchSnapshot()
    })
})