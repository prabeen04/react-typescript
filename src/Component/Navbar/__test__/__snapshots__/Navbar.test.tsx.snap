// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyNavbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Navbar
        className="my-nav transparent-bg"
        data-test="Navbar"
        expand="md"
        tag="nav"
      >
        <NavLink
          className="navbar-brand logo"
          to="/"
        >
          Typescript
        </NavLink>
        <NavbarToggler
          className="toggler"
          onClick={[Function]}
          tag="button"
          type="button"
        />
        <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          navbar={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
        >
          <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active-menu"
                className="nav-link nav-menu"
                to="/counter"
              >
                Counter
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active-menu"
                className="nav-link nav-menu"
                to="/todo"
              >
                Todo
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active-menu"
                className="nav-link nav-menu"
                to="/fetch"
              >
                Users
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavLink
            className="navbar-brand logo"
            to="/"
          >
            Typescript
          </NavLink>,
          <NavbarToggler
            className="toggler"
            onClick={[Function]}
            tag="button"
            type="button"
          />,
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/counter"
                >
                  Counter
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/todo"
                >
                  Todo
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/fetch"
                >
                  Users
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>,
        ],
        "className": "my-nav transparent-bg",
        "data-test": "Navbar",
        "expand": "md",
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Typescript",
            "className": "navbar-brand logo",
            "to": "/",
          },
          "ref": null,
          "rendered": "Typescript",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "toggler",
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/counter"
                >
                  Counter
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/todo"
                >
                  Todo
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/fetch"
                >
                  Users
                </NavLink>
              </NavItem>
            </Nav>,
            "enter": true,
            "exit": true,
            "in": false,
            "isOpen": false,
            "mountOnEnter": false,
            "navbar": true,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 350,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/counter"
                  >
                    Counter
                  </NavLink>
                </NavItem>,
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/todo"
                  >
                    Todo
                  </NavLink>
                </NavItem>,
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/fetch"
                  >
                    Users
                  </NavLink>
                </NavItem>,
              ],
              "className": "ml-auto",
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/counter"
                  >
                    Counter
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active-menu",
                    "children": "Counter",
                    "className": "nav-link nav-menu",
                    "to": "/counter",
                  },
                  "ref": null,
                  "rendered": "Counter",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/todo"
                  >
                    Todo
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active-menu",
                    "children": "Todo",
                    "className": "nav-link nav-menu",
                    "to": "/todo",
                  },
                  "ref": null,
                  "rendered": "Todo",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/fetch"
                  >
                    Users
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active-menu",
                    "children": "Users",
                    "className": "nav-link nav-menu",
                    "to": "/fetch",
                  },
                  "ref": null,
                  "rendered": "Users",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navbar
          className="my-nav transparent-bg"
          data-test="Navbar"
          expand="md"
          tag="nav"
        >
          <NavLink
            className="navbar-brand logo"
            to="/"
          >
            Typescript
          </NavLink>
          <NavbarToggler
            className="toggler"
            onClick={[Function]}
            tag="button"
            type="button"
          />
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/counter"
                >
                  Counter
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/todo"
                >
                  Todo
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active-menu"
                  className="nav-link nav-menu"
                  to="/fetch"
                >
                  Users
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavLink
              className="navbar-brand logo"
              to="/"
            >
              Typescript
            </NavLink>,
            <NavbarToggler
              className="toggler"
              onClick={[Function]}
              tag="button"
              type="button"
            />,
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                className="ml-auto"
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/counter"
                  >
                    Counter
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/todo"
                  >
                    Todo
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/fetch"
                  >
                    Users
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>,
          ],
          "className": "my-nav transparent-bg",
          "data-test": "Navbar",
          "expand": "md",
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Typescript",
              "className": "navbar-brand logo",
              "to": "/",
            },
            "ref": null,
            "rendered": "Typescript",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "toggler",
              "onClick": [Function],
              "tag": "button",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appear": false,
              "children": <Nav
                className="ml-auto"
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/counter"
                  >
                    Counter
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/todo"
                  >
                    Todo
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active-menu"
                    className="nav-link nav-menu"
                    to="/fetch"
                  >
                    Users
                  </NavLink>
                </NavItem>
              </Nav>,
              "enter": true,
              "exit": true,
              "in": false,
              "isOpen": false,
              "mountOnEnter": false,
              "navbar": true,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 350,
              "unmountOnExit": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      activeClassName="active-menu"
                      className="nav-link nav-menu"
                      to="/counter"
                    >
                      Counter
                    </NavLink>
                  </NavItem>,
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      activeClassName="active-menu"
                      className="nav-link nav-menu"
                      to="/todo"
                    >
                      Todo
                    </NavLink>
                  </NavItem>,
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      activeClassName="active-menu"
                      className="nav-link nav-menu"
                      to="/fetch"
                    >
                      Users
                    </NavLink>
                  </NavItem>,
                ],
                "className": "ml-auto",
                "navbar": true,
                "tag": "ul",
                "vertical": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active-menu"
                      className="nav-link nav-menu"
                      to="/counter"
                    >
                      Counter
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active-menu",
                      "children": "Counter",
                      "className": "nav-link nav-menu",
                      "to": "/counter",
                    },
                    "ref": null,
                    "rendered": "Counter",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active-menu"
                      className="nav-link nav-menu"
                      to="/todo"
                    >
                      Todo
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active-menu",
                      "children": "Todo",
                      "className": "nav-link nav-menu",
                      "to": "/todo",
                    },
                    "ref": null,
                    "rendered": "Todo",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active-menu"
                      className="nav-link nav-menu"
                      to="/fetch"
                    >
                      Users
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active-menu",
                      "children": "Users",
                      "className": "nav-link nav-menu",
                      "to": "/fetch",
                    },
                    "ref": null,
                    "rendered": "Users",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
